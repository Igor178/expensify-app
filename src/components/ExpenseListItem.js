import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

const ExpenseListItem = ({ description, amount, createdAt, id }) => {
    return (
        <div>
            <Link to={`/edit/${id}`}><h3>{description}</h3></Link>
            <p>{amount}$ - {createdAt}</p>
        </div>
    )
}

ExpenseListItem.propTypes = {
    description: PropTypes.string,
    amount: PropTypes.number,
    createdAt: PropTypes.number,
    // id: PropTypes.string,
}

export default ExpenseListItem


