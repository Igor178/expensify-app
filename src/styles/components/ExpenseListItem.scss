.list-item {
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 100%;
    padding: 1.3rem;
    text-decoration: none;
    color: #333;
    border: 1px solid #e5e5e5;
    border-top: none;
    transition: background .3s ease;
}

.list-item:hover {
    background-color: #f7f7f7
}

.list-item__description {
    margin-bottom: 10px;
}

.list-item__date {
    color: #666;
    opacity: 0.8;
}