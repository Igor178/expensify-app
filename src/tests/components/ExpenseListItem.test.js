import React from 'react'
import ExpenseListItem from '../../components/ExpenseListItem'
import { shallow } from 'enzyme'
import expenses from '../fixtures/expenses'

test('Render ExpenseListItem with its values', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>)
    expect(wrapper).toMatchSnapshot()
})