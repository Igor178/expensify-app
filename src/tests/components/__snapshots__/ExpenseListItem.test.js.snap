// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ExpenseListItem with its values 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      Rent
    </h3>
  </Link>
  <p>
    50
    $ - 
    0
  </p>
</div>
`;
